containers/Reports/__tests__/ReportCreateMessage.test.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import { fireEvent, render, screen } from '@testing-library/react-native'
+
+import '@testing-library/jest-native/extend-expect'
+
+import { testIdeaAdmin, testText } from '../../../tests/TestData'
+import { ReportCreateMessage } from '../ReportCreateMessage'
+
+test('Test ReportCreateMessage Snapshot', () => {
+  const route = { params: {
+    content_type: testIdeaAdmin.content_type,
+    object_pk: testIdeaAdmin.pk
+  }}
+  const { toJSON } = render(<ReportCreateMessage route={route}/>)
+  expect(toJSON()).toMatchSnapshot()
+})
+
+test('Test ReportCreateMessage, no charactures error message', async () => {
+  const route = { params: {
+    content_type: testIdeaAdmin.content_type,
+    object_pk: testIdeaAdmin.pk
+  }}
+  const description = ''
+  render(<ReportCreateMessage route={route} description={description}/>)
+  const submitButton = screen.getByLabelText('Submit')
+  fireEvent(submitButton, 'press')
+  const validationErrors = await screen.findAllByText(
+    /Please add a few words explaining why you are reporting this content/
+  )
+  expect(validationErrors).toBeTruthy()
+})
+
+test('Test ReportCreateMessage, too many charectures error message', async () => {
+  const route = { params: {
+    content_type: testIdeaAdmin.content_type,
+    object_pk: testIdeaAdmin.pk
+  }}
+  const description = testText.Lorum1039
+  render(<ReportCreateMessage route={route} description={description}/>)
+  const submitButton = screen.getByLabelText('Submit')
+  fireEvent(submitButton, 'press')
+  const validationErrors = await screen.queryAllByText(
+    /Message must be no longer then 1024 characters/
+  )
+  expect(validationErrors).toBeTruthy()
+})
+
+// button not disabled
+// test('Test ReportCreateMessage, no charactures button disabled', async () => {
+//   const route = { params: {
+//     content_type: testIdeaAdmin.content_type,
+//     object_pk: testIdeaAdmin.pk
+//   }}
+//   const description = ''
+//   render(<ReportCreateMessage route={route} description={description}/>)
```

Comment:
Do i need to add the `isValid` logic here to ensure button is disabled? I thought adding condition that would disable it should be enough but also feel I haven't quite got the testing stuff

Reply:
yes that should be it? the isValid is set by Formik i guess, so you can't change it from "outside"

Reply:
Yeah I thought that but tried tried a number of other functions then eventually console logged the button props and it's not actually disabled, but now thinking it's because I don't 'fill in' the form, maybe I have to fire an event for that, will give it a last try